src/axom/primal/geometry/Line.hpp

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_PRIMAL_LINE_HPP_
+#define AXOM_PRIMAL_LINE_HPP_
+
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Segment.hpp"
+#include "axom/primal/geometry/Vector.hpp"
+
+#include "axom/slic/interface/slic.hpp"
+
+#include <ostream>
+
+namespace axom
+{
+namespace primal
+{
+// Forward declare the templated classes and operator functions
+template <typename T, int NDIMS>
+class Line;
+
+/*!
+ * \brief Overloaded output operator for lines
+ */
+template <typename T, int NDIMS>
+std::ostream& operator<<(std::ostream& os, const Line<T, NDIMS>& line);
+
+/*!
+ * \class Line
+ *
+ * \brief Represents a line, \f$ L(t) \in \mathcal{R}^d \f$ , defined by an
+ *  origin point, \f$ P \f$ and a normalized direction vector, \f$ \vec{d} \f$,
```

Comment:
Not necessarily, but the `primal::Ray` class does too, and I can come up with a couple reasons why it could be useful. For example, the magntude doesn't have a useful meaning inherently, and making it a unit vector makes the math objects have a consistent parameterization, which is more useful in algortihms.